### PR TITLE
update links to Discord

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

 switching to discord

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @akihironitta @justusschock @awaelchli

### CODE REVIEW

**Comment by awaelchli** [comment]
@Borda Discord invite links expire after 7 days. Is this a permanent one? @lantiga created a permanent invite link last week and it looks like this: https://discord.gg/VptPCZkGNa
It is different from the one you have, could you please double check it? [See internal slack message.](https://lightning-ai-corp.slack.com/archives/C03D2J017QV/p1678186062170019?thread_ts=1678183600.838619&cid=C03D2J017QV)

**Comment by Borda** [comment]
> Discord invite links expire after 7 days. Is this a permanent one? @lantiga created a permanent invite link last week and it looks like this: [discord.gg/VptPCZkGNa](https://discord.gg/VptPCZkGNa)

thank you, fixed :)

**Review by awaelchli — APPROVED** [review]

File: .github/BECOMING_A_CORE_CONTRIBUTOR.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Lightning should reflect the broader community it serves. As such we should have
 
 ### Community
 
-We have an active [Slack](https://www.pytorchlightning.ai/community) community, where questions are asked daily.
+We have an active [Discord](https://discord.gg/VptPCZkGNa) and [Slack](https://www.pytorchlightning.ai/community) community, where questions are asked daily.
```
Comment (lantiga) [inline .github/BECOMING_A_CORE_CONTRIBUTOR.md:54]:
```suggestion
We have an active [Discord](https://discord.gg/VptPCZkGNa) community, where questions are asked daily.
```

Comment (lantiga) [reply .github/BECOMING_A_CORE_CONTRIBUTOR.md:54]: Let's remove Slack. The people that are in Slack today won't check back here anyway

**Review by lantiga — CHANGES_REQUESTED** [review]

**Review by lantiga — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by carmocca — COMMENTED** [review]
CI failure here is real. Do not force merge it yet

File: docs/source-app/get_started/go_beyond_training.rst [hunk 2]
```diff
@@ -10,9 +10,6 @@ Start from an ML system template
 
 **Goal:** We'll walk you through the 4 key steps to run a Lightning App that trains and demos a model.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/get_started/go_beyond_training.rst:13]:
```suggestion
```

File: docs/source-app/get_started/jumpstart_from_app_gallery.rst [hunk 3]
```diff
@@ -10,9 +10,6 @@ Anyone can build Apps for their own use cases and promote them on the `App Galle
 
 In return, you can benefit from the work of others and get started faster by re-using a ready-to-run App close to your own use case.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/get_started/jumpstart_from_app_gallery.rst:13]:
```suggestion
```

File: docs/source-app/get_started/jumpstart_from_component_gallery.rst [hunk 4]
```diff
@@ -10,9 +10,6 @@ Anyone can build components for their own use case and promote them on the `Comp
 
 In return, you can benefit from the work of others and add new functionalities to your Apps with minimal effort.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/get_started/jumpstart_from_component_gallery.rst:13]:
```suggestion
```

File: docs/source-app/levels/expert/index.rst [hunk 5]
```diff
@@ -10,9 +10,6 @@
 Expert skills
 #############
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (carmocca) [inline docs/source-app/levels/expert/index.rst:13]:
```suggestion
```

File: docs/source-app/get_started/training_with_apps.rst [hunk 6]
```diff
@@ -10,9 +10,6 @@ Evolve a model into an ML system
 
 **Goal:** We'll walk you through the two key steps to build your first Lightning App from your existing PyTorch Lightning scripts.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/get_started/training_with_apps.rst:13]:
```suggestion
```

File: docs/source-app/get_started/what_app_can_do.rst [hunk 7]
```diff
@@ -8,9 +8,6 @@ Discover what Lightning Apps can do in 5 min
 
 Lightning Apps can be plenty things, and while a picture is worth a thousand words, videos showing you examples should be worth even more.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/get_started/what_app_can_do.rst:11]:
```suggestion

```

File: docs/source-app/levels/advanced/index.rst [hunk 8]
```diff
@@ -16,9 +16,6 @@ Advanced skills
 ###############
 Learn to build nested components with advanced functionality.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/levels/advanced/index.rst:19]:
```suggestion
```

File: docs/source-app/levels/basic/build_a_lightning_component.rst [hunk 9]
```diff
@@ -6,9 +6,6 @@ Level 1: Package code in a lightning component
 
 **Goal:** In this guide you'll learn to develop `a Lightning component <https://lightning.ai/components>`_.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/levels/basic/build_a_lightning_component.rst:9]:
```suggestion
```

File: docs/source-app/levels/basic/index.rst [hunk 10]
```diff
@@ -16,9 +16,6 @@ Learn to package your code into Lightning components which can plug into your ex
 A Lightning component organizes arbitrary code so it can run on the cloud, manages
 its own infrastructure, cloud costs, networking and more.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/levels/basic/index.rst:19]:
```suggestion
```

File: docs/source-app/levels/expert/index.rst (hunk 5 above)
Comment (awaelchli) [inline docs/source-app/levels/expert/index.rst]:
```suggestion
```

File: docs/source-app/levels/intermediate/index.rst [hunk 11]
```diff
@@ -17,9 +17,6 @@ Intermediate skills
 ###################
 Learn to coordinate 2+ components into workflows which we call Lightning apps.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-app/levels/intermediate/index.rst:20]:
```suggestion
```

File: docs/source-pytorch/common_usecases.rst [hunk 12]
```diff
@@ -6,9 +6,6 @@ Common Workflows
 
 Customize and extend Lightning for things like custom hardware or distributed strategies.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/common_usecases.rst:9]:
```suggestion
```

File: docs/source-pytorch/expertise_levels.rst [hunk 13]
```diff
@@ -4,10 +4,6 @@ Level up
 ========
 Learn enough Lightning to match the level of expertise required by your research or job.
 
-.. join_slack::
-   :align: left
-   :margin: 30
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/expertise_levels.rst:7]:
```suggestion
```

File: docs/source-pytorch/levels/advanced.rst [hunk 14]
```diff
@@ -5,9 +5,6 @@ Advanced skills
 
 Configure all aspects of Lightning for advanced usecases.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/levels/advanced.rst:8]:
```suggestion
```

File: docs/source-pytorch/levels/core_skills.rst [hunk 15]
```diff
@@ -4,9 +4,6 @@ Basic skills
 ############
 Learn the basics of model development with Lightning. Researchers and machine learning engineers should start here.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/levels/core_skills.rst:7]:
```suggestion
```

File: docs/source-pytorch/levels/expert.rst [hunk 16]
```diff
@@ -5,9 +5,6 @@ Expert skills
 
 Customize and extend Lightning for things like custom hardware or distributed strategies.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/levels/expert.rst:8]:
```suggestion
```

File: docs/source-pytorch/levels/intermediate.rst [hunk 17]
```diff
@@ -5,9 +5,6 @@ Intermediate skills
 
 Learn to scale up your models and enable collaborative model development at academic or industry research labs.
 
-.. join_slack::
-   :align: left
-
 ----
```
Comment (awaelchli) [inline docs/source-pytorch/levels/intermediate.rst:8]:
```suggestion
```

**Pull request merged**